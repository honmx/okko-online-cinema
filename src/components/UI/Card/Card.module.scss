@import "../../../styles/variables.scss";

.cardWrapper {
  position: relative;
  transition: scale 0.15s ease;
  height: 100%;

  &:hover {
    scale: 1.05;
    z-index: 10000000;
  }
}

.cardWithTrailer {
  margin-bottom: -40px;
  background-color: $primaryDarkGreyColor;
  border-radius: 25px;
  // height: 100%;

  & .card {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
  }
}

.card {
  // height: 100%;
  width: min(25vw, 350px);
  overflow: hidden;
  border-radius: 25px;

  video {
    object-fit: cover;
  }

}

@media screen and (max-width: 960px) {
  .card, .movieInfo {
    width: min(65vw, 300px);
  }
}

@media screen and (max-width: 375px) {
  .card, .movieInfo {
    width: min(85vw, 290px);
  }
}

.img {
  display: block;
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.movieTitle {
  color: $accentWhiteFontColor;
}

.sound {
  position: absolute;
  bottom: 2px;
  right: 2px;

  &:hover {
    &>div {
      &>div {
        background-color: $primaryDarkGreyColor !important;
      }
    }
  }

  img {
    width: 14px;
    height: 14px;
  }
}

.movieInfo {
  position: absolute;
  top: 100%;
  left: 0;
  right: 0;
  padding: 5px 20px;
  background-color: $primaryDarkGreyColor;
  border-bottom-left-radius: 25px;
  border-bottom-right-radius: 25px;
}

.upperLine {
  display: flex;
  align-items: center;
}

.lowerLine {
  display: flex;
  align-items: center;
  margin-top: 3px;
  color: $primaryFontColor;
  font-size: 12px;

  &>*:not(:last-child) {
    margin-right: 10px;
  }
}

.ratingContainer {
  display: inline-flex;
  justify-content: center;
  align-items: center;
  margin-right: 10px;

  .rating {
    padding: 2px 4px;
    color: $accentWhiteFontColor;
    border-radius: 2px;
    font-size: 9px;
  }

  .usualRate {
    background-color: $primaryGreyColor;
  }

  .greenRate {
    background-color: $greenColor;
  }
}