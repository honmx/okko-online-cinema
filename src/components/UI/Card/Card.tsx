import React, { FC, useRef, useState } from "react";
import Link from "next/link";
import Image from "next/image";
import { useTranslation } from 'react-i18next';
import Button from "../Button/Button";
import Title from "../Title/Title";
import { IMovie } from "../../../types/IMovie";
import { useHover } from "../../../hooks/useHover";
import { useDelay } from "../../../hooks/useDelay";
import { isMovieType } from "../../../helpers/isMovieType";
import { capitalize } from "../../../helpers/capitalize";
import { IGenre } from "../../../types/IGenre";
import soundDisabled from "../../../assets/soundDisabled.svg";
import sound from "../../../assets/sound.svg";
import s from "./Card.module.scss";

interface Props {
  item: IGenre | IMovie;
  linkHref: string;
  ar?: 1 | 1.77 | 0.66;
  onClick?: (item: IGenre) => void;
}


const Card: FC<Props> = ({ item, linkHref, ar = 1.77, onClick }) => {

  const { i18n } = useTranslation();
  const lang = i18n.language;

  const ref = useRef<HTMLDivElement>(null);

  const isHover = useHover(ref);
  const isActive = useDelay(500, isHover);

  const [activeSound, setActiveSound] = useState<boolean>(true);

  const handleSoundClick = () => {
    setActiveSound(prev => !prev);
  }

  const handleGenreClick = () => {
    if (!onClick) return;

    onClick(item as IGenre);
  }

  return (
    <div className={`${s.cardWrapper} ${isHover && isActive && isMovieType(item) ? s.cardWithTrailer : ""}`} ref={ref} onClick={onClick && handleGenreClick}>
      {
        <Link href={linkHref} className={s.link}>
          <div className={s.card} style={{ aspectRatio: ar }}>
            {
              isHover && isActive && isMovieType(item) &&
              <video src="/trailer.mp4" autoPlay width="100%" height="100%" muted={activeSound} loop className={s.video} />
            }
            {
              (!isHover || !isActive) && isMovieType(item) &&
              <Image
                src={ar === 0.66 ? item.verticalPhoto : item.horizontalPhoto}
                alt={item.title.toString()}
                priority
                width={1920}
                height={1080}
                className={s.img}
              />
            }
            {
              !isMovieType(item) &&
              <p className={s.genreTitle}>{lang === "en" && item.originalTitle ? item.originalTitle : item.title}</p>
            }
          </div>
        </Link>
      }
      {
        isHover && isActive && isMovieType(item) && <>
          <div className={s.movieInfo}>
            <div className={s.upperLine}>
              <div className={s.ratingContainer}>
                <p className={`${s.rating} ${item.rate > 7 ? s.greenRate : s.usualRate}`}>{item.rate}</p>
              </div>
              <Title fs="14px" fw={400}>{lang === "en" && item.originalTitle ? item.originalTitle : item.title}</Title>
            </div>
            <div className={s.lowerLine}>
              <p className={s.genre}>{capitalize(lang === "en" && item.genres[0].originalTitle ? item.genres[0].originalTitle : item.genres[0].title)}</p>
              <p className={s.year}>{item.yearTill}</p>
              <p className={s.country}>{item.country.split(", ")[0]}</p>
            </div>
            <Button shape="circle" p="5px" img={!activeSound ? sound : soundDisabled} onClick={handleSoundClick} className={s.sound} />
          </div>
        </>
      }
    </div>
  )
};

export default Card;