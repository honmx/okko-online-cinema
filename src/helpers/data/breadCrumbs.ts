export const moviesPageBreadCrumbs = [
  {
    value: "Main",
    href: "/"
  },
  {
    value: "Movies",
    href: "/movies"
  }
];