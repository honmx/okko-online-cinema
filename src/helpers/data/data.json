{
  "movie": [
    {
      "id": "1",
      "title": "Время первых",
      "description": "Они шагнут в холодную неизвестность вопреки всему. Грандиозная история о первом выходе человека в открытый космос",
      "language_audio": ["Rus"],
      "language_subtitles": [],
      "duration": "2 ч 14 мин",
      "year": "2017",
      "genre": [
        "Приключения"
      ],
      "age_category": "6+",
      "director": "Дмитрий Киселёв",
      "rating": "7.9",
      "actors": [
        "Евгений Миронов",
        "Константин Хабенский",
        "Владимир Ильин"
      ],
      "image": "https://static.okko.tv/images/v2/7945999"
    },
    {
      "id": "2",
      "title": "Проксима",
      "description": "Оставить семью на Земле, чтобы отправиться в космос. Ева Грин и Мэтт Диллон на Байконуре слушают стихи Мандельштама",
      "language_audio": ["Rus"],
      "language_subtitles": ["Rus", "Eng"],
      "duration": "1 ч 47 мин",
      "year": "2019",
      "genre": [
        "Боевики"
      ],
      "age_category": "12+",
      "director": "Алис Винокур",
      "rating": "6.7",
      "actors": [
        "Ева Грин",
        "Зели Булан",
        "Мэтт Диллон"
      ],
      "image": "https://static.okko.tv/images/v2/9059097"
    },
    {
      "id": "3",
      "title": "Большое космическое путишествие",
      "description": "Трое подростков из разных уголков Советского Союза отправляются в космос, где оказываются без присмотра взрослыми",
      "language_audio": ["Rus"],
      "language_subtitles": [],
      "duration": "1 ч 2 мин",
      "year": "1975",
      "genre": [
        "Советское"
      ],
      "age_category": "12+",
      "director": "Валентин Селиванов",
      "rating": "7.1",
      "actors": [
        "Мила Берлинская",
        "Сергей Образов",
        "Игорь Сахаров"
      ],
      "image": "https://static.okko.tv/images/v2/19664744"
    },
    {
      "id": "4",
      "title": "Кин-дза-дза!",
      "description": "По дороге в булочную два советских человека попадают на другую планету. Комедийная антиутопия Георгия Данелии",
      "language_audio": ["Rus"],
      "language_subtitles": [],
      "duration": "2 ч 7 мин",
      "year": "1986",
      "genre": [
        "Фантастика"
      ],
      "age_category": "0+",
      "director": "Георгий Данелия",
      "rating": "8.3",
      "actors": [
        "Станислав Любшин",
        "Евгений Леонов",
        "Юрий Яковлев"
      ],
      "image": "https://static.okko.tv/images/v2/14810478"
    },
    {
      "id": "5",
      "title": "Первые на луне",
      "description": "Совершенно секретно: первый советский человек отправился на Луну еще в 1930-е. Фильм-мистификация и призер Венеции",
      "language_audio": ["Rus"],
      "language_subtitles": [],
      "duration": "1 ч 11 мин",
      "year": "2005",
      "genre": [
        "Фантастика"
      ],
      "age_category": "12+",
      "director": "Алексей Федорченко",
      "rating": "6.8",
      "actors": [
        "Борис Власов",
        "Виктория Ильинская",
        "Анатолий Отраднов"
      ],
      "image": "https://static.okko.tv/images/v2/18762371"
    },
    {
      "id": "6",
      "title": "Можно только представить",
      "description": "Песня, ставшая легендой: история хита, подарившего надежду миллионам",
      "language_audio": ["Rus"],
      "language_subtitles": [],
      "duration": "1 ч 45 мин",
      "year": "2018",
      "genre": [
        "Драмы"
      ],
      "age_category": "16+",
      "director": "Эндрю Эрвин",
      "rating": "7.9",
      "actors": [
        "Деннис Куэйд",
        "Дж. Майкл Финли",
        "Броди Роуз"
      ],
      "image": "https://static.okko.tv/images/v2/20197070"
    },
    {
      "id": "7",
      "title": "Принцеса Монако",
      "description": "Грэйс Келли выбирает между княжеством и новой ролью у Хичкока. Байопик голливудской звезды с Николь Кидман",
      "language_audio": ["Rus"],
      "language_subtitles": [],
      "duration": "1 ч 38 мин",
      "year": "2014",
      "genre": [
        "Драмы"
      ],
      "age_category": "16+",
      "director": "Оливье Даан",
      "rating": "6.8",
      "actors": [
        "Николь Кидман",
        "Майло Вентимилья",
        "Тим Рот"
      ],
      "image": "https://static.okko.tv/images/v2/19427951"
    },
    {
      "id": "8",
      "title": "Империя соблазна Джобс",
      "description": "Бунтарь-хиппи становится лидером IT-индустрии. Эштон Кутчер в роли основателя Apple и гениального бизнесмена",
      "language_audio": ["Rus", "Eng"],
      "language_subtitles": ["Rus", "Eng"],
      "duration": "2 ч 2 мин",
      "year": "2013",
      "genre": [
        "Драмы"
      ],
      "age_category": "12+",
      "director": "Джошуа Майкл Штерн",
      "rating": "6.3",
      "actors": [
        "Эштон Кутчер",
        "Дермот Малруни",
        "Джош Гад"
      ],
      "image": "https://static.okko.tv/images/v2/19425734"
    }
  ]
}