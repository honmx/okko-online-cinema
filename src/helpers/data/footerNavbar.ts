export const footerNavbar = [
  {
    title: "Okko",
    links: [
      {
        name: "О нас",
        to: "/about",
      },
      {
        name: "Блог",
        to: "/blog",
      },
      {
        name: "Карьера в Okko",
        to: "/career",
      },
      {
        name: "Агенты Okko",
        to: "/agents",
      },
    ]
  },
  {
    title: "Помощь",
    links: [
      {
        name: "Вопросы и ответы",
        to: "/faq",
      },
      {
        name: "Список устройств",
        to: "/devices",
      },
      {
        name: "Дистрибьюторам",
        to: "/distributors",
      },
      {
        name: "Контакты",
        to: "/contacts",
      },
    ]
  },
  {
    title: "Другое",
    links: [
      {
        name: "Акции и предложения",
        to: "/promotion",
      },
      {
        name: "Сертификаты",
        to: "/certificate",
      },
    ]
  },
];

