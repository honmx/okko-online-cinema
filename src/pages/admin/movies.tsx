import React, { FC, useEffect, useState } from "react";
import s from "./AdminMovies.module.scss";
import { GetStaticProps, NextPage } from "next";
import entitiesService from "@/services/entitiesService";
import { IMovie } from "@/types/IMovie";
import Image from "next/image";
import AdminMovieCard from "@/components/AdminMovieCard/AdminMovieCard";
import InputField from "@/components/UI/InputField/InputField";
import { useDebounce } from "@/hooks/useDebounce";

interface Props {
  movies: IMovie[];
}

const AdminMovies: NextPage<Props> = ({ movies }) => {

  
  const [filteredMovies, setFilteredMovies] = useState<IMovie[]>(movies);
  const [value, setValue] = useState<string>("");
  
  const debouncedValue = useDebounce(value);
  
  useEffect(() => {
    setFilteredMovies(movies.filter(movie =>
      movie.title.toLowerCase().includes(value)
      || movie.originalTitle?.toLowerCase().includes(value)));
  }, [debouncedValue]);

  const handleChange = (value: string) => {
    setValue(value);
  }

  return (
    <div className={s.adminMoviesPageContainer}>
      <InputField type="text" value={value} placeholder="Название фильма" onChange={handleChange} appearanceType="transparent" className={s.input} />
      <div className={s.moviesContainer}>
        {
          filteredMovies.map(movie => <AdminMovieCard key={movie.id} movie={movie} />)
        }
      </div>
    </div>
  )
};

export const getStaticProps: GetStaticProps = async () => {

  const movies = await entitiesService.getAdminMovies();

  return {
    props: {
      movies: movies.filter(movie => movie.horizontalPhoto)
    }
  }
}

export default AdminMovies;
