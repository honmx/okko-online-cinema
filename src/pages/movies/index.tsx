import React, { ReactNode, useEffect, useState } from "react";
import { NextPageWithLayout } from "@/types/NextPageWithLayout";
import MoviesPageLayout from "@/components/MoviesPageLayout/MoviesPageLayout";
import Carousel from "@/components/UI/Carousel/Carousel";
import Card from "@/components/UI/Card/Card";
import { IMovie } from "@/types/IMovie";
import { GetStaticProps, GetStaticPropsResult } from "next";
import axios from "axios";
import Head from "next/head";
import s from "./Movies.module.scss";
import { useAppDispatch } from "@/store/hooks";
import { clearFilters, setSelectedMinRating, setSelectedGenre, setSelectedActor, setSelectedCountry } from "@/store/slices/moviesFilterSlice";
import { IGenre } from "@/types/IGenre";
import { useSelectedFilters } from "@/hooks/useSelectedFilters";
import { useRouter } from "next/router";
import dynamic from "next/dynamic";
import { areFiltersClear } from "@/helpers/areFiltersClear";
import entitiesService from "@/services/entitiesService";;
import { serverSideTranslations } from "next-i18next/serverSideTranslations";

interface Props {
  movies: IMovie[];
  bestMovies: IMovie[];
  russianMovies: IMovie[];
}

const ClientCarousel = dynamic(() => import("../../components/UI/Carousel/Carousel"), {
  ssr: false,
  loading: () => <p>loading...</p>
});

const Movies: NextPageWithLayout<Props> = ({ movies, bestMovies, russianMovies }) => {

  const dispatch = useAppDispatch();
  const router = useRouter();

  const {
    selectedGenre,
    selectedCountry,
    selectedMinRating,
    selectedMinCountOfRating,
    selectedProducer,
    selectedActor,
    selectedSortBy,
  } = useSelectedFilters();

  useEffect(() => {
    if (areFiltersClear({
      selectedGenre, selectedCountry, selectedMinRating,
      selectedMinCountOfRating, selectedProducer, selectedActor, selectedSortBy
    })) return;

    router.push("/movies/filters");
  }, [
    JSON.stringify({
      selectedGenre, selectedCountry,
      selectedMinRating, selectedMinCountOfRating,
      selectedProducer, selectedActor, selectedSortBy
    })
  ]);

  return (
    <>
      <Head>
        <title>Фильмы</title>
        <meta
          name="description"
          content="Смотреть фильмы онлайн в хорошем качестве"
        />
      </Head>
      <div className={s.moviesContainer}>
        <div className={s.block}>
          <ClientCarousel
            title="Лучшее"
            linkHref="/movies/filters"
            onTitleClick={() => dispatch(setSelectedMinRating(8.8))}
            className={s.carousel}
          >
            {
              bestMovies
                .map(movie => <Card key={movie.id} item={movie} linkHref={`/movie/${movie.title}`} />)
            }
          </ClientCarousel>
          <ClientCarousel
            title="Российские Фильмы"
            linkHref="/movies/filters"
            onTitleClick={() => dispatch(setSelectedCountry("Россия"))}
            className={s.carousel}
          >
            {
              russianMovies
                .map(movie => <Card key={movie.id} item={movie} linkHref={`/movie/${movie.title}`} />)
            }
          </ClientCarousel>
        </div>
      </div>
    </>
  )
};

Movies.getLayout = (page: ReactNode) => {
  return (
    <MoviesPageLayout>{page}</MoviesPageLayout>
  )
}

export const getStaticProps: GetStaticProps = async (context): Promise<GetStaticPropsResult<Record<string, unknown>>> => {

  const movies = await entitiesService.getMovies();

  const bestMovies = await entitiesService.getMoviesByMinRating(8.8);
  const russianMovies = await entitiesService.getMoviesByCountry("Россия");

  return {
    props: {
      movies: movies.filter(movie => movie.horizontalPhoto),
      bestMovies: bestMovies.filter(movie => movie.horizontalPhoto),
      russianMovies: russianMovies.filter(movie => movie.horizontalPhoto),
      ...(await serverSideTranslations(context.locale as string, [
        "common",
        "header",
        "footer",
        "moviesPage"
      ])),
    }
  }
}

export default Movies;