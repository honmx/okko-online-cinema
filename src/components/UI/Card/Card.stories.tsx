import { Meta, StoryObj } from "@storybook/react";
import Card from "./Card";
import { movie } from "../../../helpers/data/data.json";

const meta: Meta<typeof Card> = {
  title: "Card",
  component: Card,
  tags: ["autodocs"],
  argTypes: {

  }
}

export default meta;

type Story = StoryObj<typeof Card>;

export const Card16x9: Story = {
  args: {
    item: movie[0],
    linkHref: "/"
  }
}

export const Card1x1: Story = {
  args: {
    item: movie[0],
    linkHref: "/",
    ar: 1
  }
}

export const Card9x16: Story = {
  args: {
    item: movie[0],
    linkHref: "/",
    ar: 0.66
  }
}