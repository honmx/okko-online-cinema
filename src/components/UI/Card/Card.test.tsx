import { fireEvent, render } from "@testing-library/react";
import Card from "./Card";
import { movies } from "@/helpers/data/movies";

describe("", () => {

  test("component renders", () => {
    const { container } = render(
      <Card item={movies[0]} linkHref={movies[0].title} />
    );
    
    expect(container.getElementsByClassName("cardWrapper")[0]).toBeInTheDocument();
  });

  test("card scales if hovered", () => {

    const { container } = render(
      <Card item={movies[0]} linkHref={movies[0].title} />
    );

    const cardWrapper = container.getElementsByClassName("cardWrapper")[0];

    fireEvent.mouseOver(cardWrapper);

    setTimeout(() => {
      expect(cardWrapper).toHaveStyle("scale: 1.05");
    }, 150);
  });
  
  test("card shows video if hovered for 1s", () => {

    const { container } = render(
      <Card item={movies[0]} linkHref={movies[0].title} />
    );

    const cardWrapper = container.getElementsByClassName("cardWrapper")[0];

    fireEvent.mouseOver(cardWrapper);

    setTimeout(() => {
      expect(cardWrapper.getElementsByClassName("video")[0]).toBeInTheDocument();
    }, 1100);
  });

  test("card with default ar", () => {
    const component = render(
      <Card item={movies[0]} linkHref={movies[0].title} />
    );

    expect(component).toMatchSnapshot();
  });

  test("card with ar = 1", () => {
    const component = render(
      <Card item={movies[0]} linkHref={movies[0].title} ar={1} />
    );

    expect(component).toMatchSnapshot();
  });

  test("card with ar = 0.66", () => {
    const component = render(
      <Card item={movies[0]} linkHref={movies[0].title} ar={0.66} />
    );

    expect(component).toMatchSnapshot();
  });
})