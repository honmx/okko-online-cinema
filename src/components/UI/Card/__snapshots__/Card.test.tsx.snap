// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` card with ar = 0.66 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cardWrapper "
      >
        <a
          class="link"
          href="1+1"
        >
          <div
            class="card"
          >
            <img
              alt="1+1"
              class="img"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="1080"
              src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2Fbf93b465-1189-4155-9dd1-cb9fb5cb1bb5%2Forig&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2Fbf93b465-1189-4155-9dd1-cb9fb5cb1bb5%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2Fbf93b465-1189-4155-9dd1-cb9fb5cb1bb5%2Forig&w=3840&q=75 2x"
              style="color: transparent;"
              width="1920"
            />
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cardWrapper "
    >
      <a
        class="link"
        href="1+1"
      >
        <div
          class="card"
        >
          <img
            alt="1+1"
            class="img"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="1080"
            src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2Fbf93b465-1189-4155-9dd1-cb9fb5cb1bb5%2Forig&w=3840&q=75"
            srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2Fbf93b465-1189-4155-9dd1-cb9fb5cb1bb5%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1946459%2Fbf93b465-1189-4155-9dd1-cb9fb5cb1bb5%2Forig&w=3840&q=75 2x"
            style="color: transparent;"
            width="1920"
          />
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[` card with ar = 1 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cardWrapper "
      >
        <a
          class="link"
          href="1+1"
        >
          <div
            class="card"
          >
            <img
              alt="1+1"
              class="img"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="1080"
              src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2F56bf76fb-58e9-42b6-94d8-9e3bd4c58e85%2Forig&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2F56bf76fb-58e9-42b6-94d8-9e3bd4c58e85%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2F56bf76fb-58e9-42b6-94d8-9e3bd4c58e85%2Forig&w=3840&q=75 2x"
              style="color: transparent;"
              width="1920"
            />
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cardWrapper "
    >
      <a
        class="link"
        href="1+1"
      >
        <div
          class="card"
        >
          <img
            alt="1+1"
            class="img"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="1080"
            src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2F56bf76fb-58e9-42b6-94d8-9e3bd4c58e85%2Forig&w=3840&q=75"
            srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2F56bf76fb-58e9-42b6-94d8-9e3bd4c58e85%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2F56bf76fb-58e9-42b6-94d8-9e3bd4c58e85%2Forig&w=3840&q=75 2x"
            style="color: transparent;"
            width="1920"
          />
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[` card with default ar 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cardWrapper "
      >
        <a
          class="link"
          href="1+1"
        >
          <div
            class="card"
          >
            <img
              alt="1+1"
              class="img"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="1080"
              src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2F56bf76fb-58e9-42b6-94d8-9e3bd4c58e85%2Forig&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2F56bf76fb-58e9-42b6-94d8-9e3bd4c58e85%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2F56bf76fb-58e9-42b6-94d8-9e3bd4c58e85%2Forig&w=3840&q=75 2x"
              style="color: transparent;"
              width="1920"
            />
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cardWrapper "
    >
      <a
        class="link"
        href="1+1"
      >
        <div
          class="card"
        >
          <img
            alt="1+1"
            class="img"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="1080"
            src="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2F56bf76fb-58e9-42b6-94d8-9e3bd4c58e85%2Forig&w=3840&q=75"
            srcset="/_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2F56bf76fb-58e9-42b6-94d8-9e3bd4c58e85%2Forig&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.mds.yandex.net%2Fget-kinopoisk-image%2F1900788%2F56bf76fb-58e9-42b6-94d8-9e3bd4c58e85%2Forig&w=3840&q=75 2x"
            style="color: transparent;"
            width="1920"
          />
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
